code/modules/religion/darkness_sect.dm

```diff
@@ -0,0 +1,133 @@
+	if(!ismovable(religious_tool))
+		CRASH("[name]'s perform_rite had a movable atom that has somehow turned into a non-movable!")
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool?.buckled_mobs?.len)
+		return FALSE
+	var/mob/living/carbon/human/human2darken = locate() in movable_reltool.buckled_mobs
+	if(!human2darken)
+		return FALSE
+	human2darken.set_species(/datum/species/shadow)
+	human2darken.visible_message("<span class='notice'>[human2darken] has been converted by the rite of [name]!</span>")
+	return ..()
+
+/datum/religion_rites/darkobelisk
+	name = "Obelisk Manifestation"
+	desc = "Creates an obelisk that generates favor when in a dark area."
+	ritual_length = 15 SECONDS
+	invoke_msg = "I summon forth an obelisk, to appease the darkness."
+	favor_cost = 100 //Sect starts with 100 favor to begin
+
+/datum/religion_rites/darkobelisk/invoke_effect(mob/living/user, atom/religious_tool)
+	var/altar_turf = get_turf(religious_tool)
+	var/obj/structure/darkobelisk/obelisk = new(altar_turf)
+	var/datum/religion_sect/darkness_sect/sect = GLOB.religious_sect
+	sect.obelisks += obelisk
+	obelisk.AddComponent(/datum/component/dark_favor, user)
+	obelisk.set_light(sect.light_reach, sect.light_power, DARKNESS_INVERSE_COLOR)
+	playsound(altar_turf, 'sound/magic/fireball.ogg', 50, TRUE)
+	return ..()
+
+/obj/structure/darkobelisk
+	name = "obelisk of darkness"
+	desc = "Grants favor from being shrouded in darkness."
+	icon = 'icons/obj/hand_of_god_structures.dmi'
+	icon_state = "darkness-obelisk"
+	density = TRUE
+
+/datum/religion_rites/extend_darkness
+	name = "Extend Darkness"
+	desc = "Grow the reach of darkness extending from the altar, and any obelisks."
+	ritual_length = 10 SECONDS
+	invoke_msg = "Darkness, reach your tendrils from my altar, and extend thy domain."
+	favor_cost = 75
+
+/datum/religion_rites/extend_darkness/invoke_effect(mob/living/user, atom/religious_tool)
+	. = ..()
+	var/datum/religion_sect/darkness_sect/sect = GLOB.religious_sect
+	if(!sect)
+		return
+	sect.light_reach += 2
+	sect.light_power -= 1
+	religious_tool.set_light(sect.light_reach, sect.light_power, DARKNESS_INVERSE_COLOR)
+	for(var/obj/structure/darkobelisk/D in sect.obelisks)
+		D.set_light(sect.light_reach, sect.light_power, DARKNESS_INVERSE_COLOR)
+
+// Favor generator component. Used on the altar and obelisks
+/datum/component/dark_favor
+	var/mob/living/creator
+
+/datum/component/dark_favor/Initialize(mob/living/L)
+	. = ..()
+	if(!L)
+		return
+	creator = L
+	START_PROCESSING(SSobj, src)
+
+/datum/component/dark_favor/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
+
+/datum/component/dark_favor/process()
+	if(!istype(parent, /atom) || !istype(creator))
+		return
+	var/atom/P = parent
+	var/turf/T = P.loc
+	if(!istype(T))
+		return
+	var/light_amount = T.get_lumcount()
+	var/favor_gained = max(1 - light_amount, 0)
+	GLOB.religious_sect?.adjust_favor(favor_gained, creator)
```

Comment:
if you want to check for sect do it earlier (maybe with istype check?)
or better tie it to signals